homeassistant/components/androidtv/media_player.py

```diff
@@ -760,7 +785,9 @@ async def async_update(self):
             self._current_app,
             running_apps,
             self._hdmi_input,
-        ) = await self.aftv.update(self._get_sources)
+        ) = await self.aftv.update(
+            self._get_sources and self._source_provider == SOURCE_PROVIDER_RUNNING_APPS
+        )
```

Comment:
You missed line 796, it should be:

```python
if running_apps and self._source_provider == SOURCE_PROVIDER_RUNNING_APPS:
```

Reply:
It would be an empty array in that scenario but sure I'll add it.